fe/fe-core/src/main/java/com/starrocks/qe/scheduler/dag/AllAtOnceExecutionSchedule.java

```diff
@@ -15,29 +15,45 @@
 package com.starrocks.qe.scheduler.dag;
 
 import com.starrocks.common.UserException;
+import com.starrocks.qe.scheduler.Coordinator;
 import com.starrocks.qe.scheduler.Deployer;
 import com.starrocks.qe.scheduler.slot.DeployState;
 import com.starrocks.rpc.RpcException;
 import com.starrocks.thrift.TUniqueId;
 
+import java.util.ArrayList;
 import java.util.List;
 
 // all at once execution schedule only schedule once.
 public class AllAtOnceExecutionSchedule implements ExecutionSchedule {
+    private Coordinator coordinator;
     private Deployer deployer;
     private ExecutionDAG dag;
 
     @Override
-    public void prepareSchedule(Deployer deployer, ExecutionDAG dag) {
+    public void prepareSchedule(Coordinator coordinator, Deployer deployer, ExecutionDAG dag) {
+        this.coordinator = coordinator;
         this.deployer = deployer;
         this.dag = dag;
     }
 
     @Override
     public void schedule() throws RpcException, UserException {
+        List<DeployState> states = new ArrayList<>();
         for (List<ExecutionFragment> executionFragments : dag.getFragmentsInTopologicalOrderFromRoot()) {
             final DeployState deployState = deployer.createFragmentExecStates(executionFragments);
             deployer.deployFragments(deployState);
+            states.add(deployState);
+        }
+
+        while (true) {
+            states = coordinator.assignIncrementalScanRangesToDeployStates(deployer, states);
```

Comment:
`schedule()` is called only once by `DefaultCoordinator::deliverExecFragments`. Will it will be called in the future after some RPC reports from BE?

Reply:
@ZiheLiu  Yes, `schedule()` will be called once.

And for `PhasedExecutionSchedule`, I put delivery of incremental scan ranges into `doDeploy`, not `schedule`.

Reply:
Got it. `AllAtOnceExecutionSchedule::schedule()` will call `assignIncrementalScanRangesToDeployStates` and `deployFragments` many times until there isn't scan ranges anymore.